### PR TITLE
validate custom urls

### PR DESCRIPTION
**Pull request by nelsonpecora**
* [trello ticket](https://trello.com/c/4bXk7e7o/10-validate-custom-urls-kiln)
* validates that a url is routable in a particular site
* allows full urls
* allows unsetting custom url (with improved status message)
* relies on https://github.com/nymag/amphora/pull/300

### CODE REVIEW

File: controllers/publish-pane.js [hunk 1]
```diff
@@ -153,6 +181,21 @@ module.exports = function () {
...
+        val = input.value,
+        routes = this.routes;
+
+      // validate that what the user typed in is routable
+      // note: if it's empty string, catch it early (removing custom urls is totally valid)
+      // note: if it's a full url, assume the user knows what they're doing and say it's valid
+      if (val === '' || val.match(/^http/i) || isValidUrl(val, routes)) {
+        input.setCustomValidity('');
+      } else {
+        input.setCustomValidity('Custom URL must match an available route!');
```
Comment (davidandrewfrey) [inline controllers/publish-pane.js]: `customValidity` refers to the message here. Is this setting something that is always a string or can it be another type, like an object?

File: controllers/publish-pane.js [hunk 2]
```diff
@@ -185,13 +228,19 @@ module.exports = function () {
         })
         .then(function () {
           progress.done('page');
-          progress.open('page', 'Saved custom page url', true);
+          if (url === '' || url === undefined) {
```
Comment (davidandrewfrey) [inline controllers/publish-pane.js]: A truthy test of `url` isn't explicit enough?

File: controllers/publish-pane.js (hunk 1 above)
Comment (nelsonpecora) [inline controllers/publish-pane.js]: it's always a string. emptystring means it's valid

File: controllers/publish-pane.js (hunk 2 above)
Comment (nelsonpecora) [inline controllers/publish-pane.js]: hmm, might be. I was being very explicit here, but I'll test it otu

**Comment by davidandrewfrey** [comment]
👍

**Pull request merged**